LiteCore/Query/SQLiteFleeceUtil.cc

```diff
@@ -325,6 +326,44 @@ namespace litecore {
         }
     }
 
+    enhanced_bool_t booleanValue(sqlite3_context* ctx, sqlite3_value *arg) {
```

Comment:
This is cool. Does this mean we can do N1QL boolean value as below?

* MISSING, NULL, and FALSE are FALSE
* numbers +0, -0, and NaN are FALSE
* empty strings, arrays, and objects are FALSE
* all other values are TRUE

Reply:
We can, but we shouldn't because that is wrong....this boolean evaluation should match server.  

>   MISSING is MISSING.
NULL is NULL.
False is false.
Numbers +0, -0, and NaN are false.
Empty strings, arrays, and objects are false.
All other values are true.